src/s2wsjson/validate_values_mixin.py

```diff
@@ -80,3 +48,28 @@ def to_dict(self: C) -> dict:
     def from_json(cls: Type[C], json_str: str) -> C:
         gen_model: C = cls.parse_raw(json_str)
         return gen_model
+
+
+def convert_to_s2exception(f : Callable):
+    def inner(*args, **kwargs):
+        try:
+            return f(*args, **kwargs)
+        except (ValidationError, TypeError) as e:
+            raise S2ValidationError(args[0], 'Pydantic raised a format validation error.', pydantic_validation_error=e)
```

Comment:
`args[0]` in the `parse_raw` and `__init__` case references the class which is created. Redundant as that class is necessary to perform the correct `from_json` or `__init__` call. Might be useful to show all args to show both offending and non-offending values. The pydantic exception already says which key is the issue and what the issue is, but not the value which was used.

With `__setattr__` case it makes sense to also include the key and the new value that caused the exception.

What do you think? Should we extend S2ValidationError by also adding all new keys and values? (They should be contained in `args` and `kwargs`)
With `__init__` case

Reply:
I agree!

Not sure if this conversion should rather live inside the exception `S2ValidationError` or in the patch function `convert_to_s2exception`.

I  suggest to create a new Issue/PR to discuss the details.

Reply:
Done, see #8!